prymer/primer3/primer3_parameters.py

```diff
@@ -145,3 +208,131 @@ def max_primer_length(self) -> int:
     def min_primer_length(self) -> int:
         """Minimum primer length."""
         return int(self.primer_sizes.min)
+
+
+@dataclass(frozen=True, init=True, slots=True)
+class Primer3Parameters(PrimerAndAmpliconParameters):
+    """A deprecated alias for `PrimerAndAmpliconParameters` intended to maintain backwards
+    compatibility with earlier releases of `prymer`."""
+
+    warnings.warn(
+        "The Primer3Parameters class was deprecated, use PrimerAndAmpliconParameters instead",
+        DeprecationWarning,
+        stacklevel=2,
+    )
+
+
+@dataclass(frozen=True, init=True, slots=True)
+class ProbeParameters:
+    """Holds common primer design options that Primer3 uses to inform internal probe design.
+
+    Attributes:
+        probe_sizes: the min, optimal, and max probe size
+        probe_tms: the min, optimal, and max probe melting temperatures
+        probe_gcs: the min and max GC content for individual probes
+        number_probes_return: the number of probes to return
+        probe_max_dinuc_bases: the max  number of bases in a dinucleotide run in a probe
+        probe_max_polyX: the max homopolymer length acceptable within a probe
+        probe_max_Ns: the max number of ambiguous bases acceptable within a probe
+        probe_max_self_any: the maximal local alignment score of aligning the probe to itself
+        probe_max_self_any_thermo: the maximal melting temperature of the most stable structure
+            resulting from aligning the probe to itself
+        probe_max_self_end: max allowable 3'-anchored global alignment score when testing a
+            probe for self-complementarity
+        probe_max_self_end_thermo: the maximal melting temperature of the most stable structure
+            resulting from aligning the 3' end of the probe
+        probe_max_hairpin_thermo: most stable monomer structure as calculated by a thermodynamic
+            approach
+        probe_excluded_region: the excluded region (start, length) that probes shall not overlap
```

Comment:
This does not feel like it belongs here - or if it does, it needs a lot more documentation on how to specify this in a target agnostic way.

Reply:
`probe_excluded_region` has been stripped out in the latest commit!